src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs

```diff
@@ -538,10 +539,14 @@ public override void Write(string format, object arg0, object arg1, object arg2)
             }
         }
 
-        public override void Write(string format, params object[] arg)
+        public override void Write(string format, params object?[] arg)
         {
             if (GetType() == typeof(StreamWriter))
             {
+                if (arg == null)
+                {
+                    throw new ArgumentNullException(nameof(arg));
+                }
```

Comment:
Added to be more consistent with argument validation across the various paths.

Reply:
Why not add validation to `format` as well?

Reply:
Updated.